File: backend/models/job.py
from pydantic import BaseModel
from typing import Optional


class Job(BaseModel):
    id: str
    title: str
    company: str
    location: str
    description: str
    url: str
    salary_min: Optional[float] = None
    salary_max: Optional[float] = None
    created: str
    category: str


class JobSearchResponse(BaseModel):
    jobs: list[Job]
    total: int
    page: int
    pages: int


class Country(BaseModel):
    code: str
    name: str


class Category(BaseModel):
    code: str
    name: str

File: frontend/src/types/job.ts
export interface Job {
  id: string
  title: string
  company: string
  location: string
  description: string
  url: string
  salary_min?: number
  salary_max?: number
  created: string
  category: string
}

export interface JobSearchResponse {
  jobs: Job[]
  total: number
  page: number
  pages: number
}

export interface Country {
  code: string
  name: string
}

export interface Category {
  code: string
  name: string
}

export interface SearchParams {
  country: string
  category: string
  keyword: string
  page: number
}
